internal/kinds/capp/resourcemanagers/ksvc.go

```diff
@@ -120,45 +124,67 @@ func (k KnativeServiceManager) isResumed(capp cappv1alpha1.Capp) bool {
 		!capp.Status.StateStatus.LastChange.IsZero()
 }
 
-// CreateOrUpdateObject ensures a KnativeService resource exists based on the provided Capp.
-// If the Capp doesn't require a KnativeService, it triggers a cleanup.
-// Otherwise, it either creates a new KnativeService or updates an existing one based on the Capp's specifications.
-func (k KnativeServiceManager) CreateOrUpdateObject(capp cappv1alpha1.Capp) error {
+// Manage creates or updates a KnativeService resource based on the provided Capp if it's required.
+// If it's not, then it cleans up the resource if it exists.
+func (k KnativeServiceManager) Manage(capp cappv1alpha1.Capp) error {
+	if k.IsRequired(capp) {
+		return k.createOrUpdate(capp)
+	}
+
+	if err := k.CleanUp(capp); err != nil {
+		return err
+	}
+
+	k.Log.Info("Disabling Capp")
```

Comment:
disabling capp should come befroe the clean up function only the event after you can maybe divide the log for trying to disable capp and disabled capp successfully